server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -3619,8 +3619,12 @@ public UserVmVO doInTransaction(TransactionStatus status) throws InsufficientCap
 
                 _vmDao.persist(vm);
                 for (String key : customParameters.keySet()) {
-                    //handle double byte strings.
-                    vm.setDetail(key, Integer.toString(Integer.parseInt(customParameters.get(key))));
+                    if( key.equalsIgnoreCase(VmDetailConstants.ROOT_DISK_CONTROLLER) || key.equalsIgnoreCase(VmDetailConstants.DATA_DISK_CONTROLLER)) {
+                        vm.setDetail(key, customParameters.get(key));
+                    } else {
+                        // handle double byte strings.
+                        vm.setDetail(key, Integer.toString(Integer.parseInt(customParameters.get(key))));
```

Comment:
In what cases would we do that @bvbharat (CLOUDSTACK-9638) ? do you think, the values set in database should be a string?